### ISSUE TITLE
User guide topics

### ISSUE THREAD

**Issue by docwilmot** [backdrop-ops/backdropcms.org#263]
Continuing from #55 
We've agreed and started on user docs
This issue will be to suggest, critique and track the topics to be included in the docs

- [x] Quick start guide
- [x] Modules themes and layouts
- [x] Nodes and Entities
- [x] Understanding Layouts
- [x] Blocks
  - [x] Configuring blocks: examples
- [x] Content types
  - [x] Create a custom content type
  - [x] Add fields to a content type
- [x] Display settings (View modes)
- [x] The Admin bar
- [x] Menus
- [x] Project installer
- [x] Taxonomy and categorizing content
- [x] Users, roles and permissions
- [x] Views basics
  - [x] Views exercises
  - [x] Configuring Views - advanced
  - [x] Configuring Views
- [ ] Installation
- [ ] Themes
  - [ ] Color
- [ ] Creating content
- [ ] Multilingual
- [ ] Configuration Management
- [ ] Multisite
- [ ] Book
- [ ] Ckeditor
- [ ] Comment
- [ ] Contact forms
- [ ] Site reports
- [ ] Fields
  - [ ] Date
  - [ ] File
  - [ ] Image
- [ ] Filters
- [ ] Updating project code
- [ ] Pathauto
- [ ] Search
- [ ] Redirects
- [ ] Howtos
  - [ ] Customizing the front page
  - [ ] Creating lists of content
  - [ ] Create an image gallery
  - [ ] Customize the User Registration Form
  - [ ] Create a Blog site
  - [ ] Create a personal site
  - [ ] Members-only site

**Comment by Graham-72** [comment]
@docwilmot How do you envisage we should structure our comments so that we can organise any debate? Could perhaps the first person to contribute a comment (or addition or amendment) on one of the above items start a new issue for that specific item? Issue #55 has served us well but has become unwieldy and it would be a shame if this new issue were to be a catch-all.

**Comment by docwilmot** [comment]
I'd say if yo have recommendations about the list itself, or the process, then post here (or start a new issue if you feel it may generate some discussion or is ultimately resolvable), but if youre critiquing a particular doc, then post a  new comment, link to this, and tag it documentation.

**Comment by tomgrandy** [comment]
@docwilmot - The api.backdropcms.org has an audience, but not the one I’m looking to narrow in on. 

The main BD User Guide: https://backdropcms.org/user-guide seems to be aimed at the site builder primarily and overlaps with skills the site user needs to know as well. 

The handbook’s Developer Guide: https://backdropcms.org/requirements (beginning here) are still needs of the Site Builder and carries into the Site Developer category

Not trying to muddy the water, but I see four distinct audiences here: 

- Evaluators who want to know what BD has to offer them, 
- Site Users who need to know the basics of how to create content in BD, 
- Site Builders who need to know how to install BD and create a site, and 
- Site Developers who have a need to know how to install, customize, convert, and upgrade a sit

e. 

Would you agree?  Are there other audiences we are not addressing?

I might be stating the obvious, but want to define the audiences first.

**Comment by docwilmot** [comment]
Seems right. What do we do next?

**Comment by tomgrandy** [comment]
Next would be to identify how to separate the four so, as they visit the backdrop.org site, they can easily identify what they need by what role they identify with.  Should there be a User Guide for a Content Editor, a Site Builder, and a Site Developer as well as an overview for Site Evaluators?  Though we may know these roles by title, would your new user be able to self-identify by role?  If not, is it worth considering labeling them something different or guiding them to an area by identifying their particular need?

**Comment by docwilmot** [comment]
The developer docs should stay on the API site in proximity to the generated API info; that seems logical. 

It may be best to guide the others by their need; particularly, we should plan the landing page at https://backdropcms.org/user-guide in a format which asks what you want to do, and link to the individual docs or subsections of the docs. Perhaps link from the landing page to sub-landing-pages as well. 

See the Wordpress landing page for example:
![351d9ae2-bf57-11e5-8fe5-f7aa04d2d881](https://cloud.githubusercontent.com/assets/1042283/18881880/a85255f0-84a2-11e6-849e-8fbf629f74a3.PNG)

**Comment by tomgrandy** [comment]
Now that I have finished up reading https://github.com/backdrop-ops/backdropcms.org/issues/55 it looks as if June 15-17 were stellar days for documentation!  Much of this seems to have been hashed out before and many iterations of documentation floating with some of it landing on B.org.

Top down, given that this is a small group and trying to ensure that any documentation that goes on the site is concise, complete, and current to the most recent version of BD, you have a conundrum on your hands:  either so simple that documentation is not helpful or so instructional that it must be kept updated to maintain relevance.

Looking through the documentation that has been created, there is a great deal of work already done and now it is a matter of how to present it in a logical fashion.

I do like the idea of starting simple and growing in complexity as you have shown above in your WP example.  Screenshots are a must, but they can get dated as well.  Videos are wonderful, but again, they get outdated.  Most of the above WP outline I would never read.

As a newb, I would probably need to know in this order:

**ADOPTER NEEDS:**

- What is BD?
- Why should I choose BD for my organization?
- What advantage is there to using BD?
- What features are included with BD on install?
- How can I extend BD?
- BD showcase...
- Video Overview

**GETTING STARTED**

- Download and Install of BD
- Converting D7 Sites to BD
- Installing Modules
- Installing Themes

**CONTENT EDITOR**

- Logging In
- Creating Content
- Admin Bar
- BD Video Lessons (Keeping it super general)

**SITE DEVELOPER / BUILDER**

- Creating Content Types
- Working with Layouts
- Blocks
- Users, Roles, and Permissions
- Taxonomy
- Menus
- Forms
- Images
- File System
- URL Alias
- URL Redirects
- Building a View
- BD Video Lessons (Very Specific)

**DESIGN/UX TEAM**

- How To Create a BD Theme (Best Practice)
- How To Create a BD Layout (Best Practice)

**CONTRIBUTORS**

- How to port a module to BD
- How to create a BD module
- BD Video Lessons

**ADVANCED DOCUMENTATION**

- Terminology
- Backdrop API
- Coding Standards
- Writing Secure Code

Just a few thoughts before eating lunch!

**Comment by wesruv** [comment]
Let me know if any graphics are needed to explain something in docs, I can
whip up some stuff.

**Comment by olafgrabienski** [comment]
> I see four distinct audiences here:
> 
> Evaluators who want to know what BD has to offer them,
> Site Users who need to know the basics of how to create content in BD,
> Site Builders who need to know how to install BD and create a site, and
> Site Developers who need to know how to install, customize, convert, and upgrade a site.

In my opinion, it's not easy to distinguish between Site Users (or Content Editors) and Site Builders. If a  prospective Content Editor (for instance a blogger) installs Backdrop him/herself, he or she may do some basic site building before posting content.

Another issue to consider: When Site Users / Content Editors use a page which was built by a Site Builder, we don't know how the site was customized. Example: the Site builder replaced the content types "Page" and "Post" with custom ones. Ideas how to manage that?

**Comment by jackaponte** [comment]
Jumping in briefly to say--as a site builder who builds sites primarily for site users/content editors who _never_ do site building, I think that the audience distinctions are important to preserve there! Luckily we can certainly anticipate and account for overlapping audiences in our approach here.

**Comment by tomgrandy** [comment]
@olafgrabienski - Like you, I am someone who wears many hats when it comes to creating new sites for my company.  So the overlap is a worthwhile concern as are customizations in naming conventions that a site builder / developer may choose for the content types.  Therefore, it is important that we keep the documentation general to a point.

For example, if we identify a Page as a Content Type and use it as the basis for describing how to create a new Content Type, it is somewhat implied that Content Types are unlimited in their customizations and naming.  Page is just one sample Content Type that is preloaded.

@jackaponte - Would you agree that the audience distinctions need to be preserved, but not necessarily named in our documentation.  As we write the documentation in a hierarchy from simple to complex, we may have the roles in mind as we write.  For example, when we are documenting the basic fields to be populated in a Page, we are considering the user as a Site User.  These audiences help us keep their distinct needs close together in the documentation to keep them from having to wade through documentation that does not pertain to them to find the next step.

Therefore, I'm proposing we that we not use the named audience type in the subtopics, but keep them in mind when grouping topics for that audience.

For example instead of:

**CONTENT EDITOR**

Logging In
Creating Content
Admin Bar

The documentation would read:

**CREATING CONTENT**

Logging In
Creating Content
Admin Bar

Though as writers of the documentation we recognize who the audience is:  Content Editors.

Make sense?

@jenlampton @docwilmot - In the past has documentation gotten bogged down in the details such as these?  Would it be better to forge ahead so there is _some_ documentation out there instead of promising that it is coming soon?

**Comment by jackaponte** [comment]
@tomgrandy, thanks for your detailed reasoning above!

> Therefore, I'm proposing we that we not use the named audience type in the subtopics, but keep them in mind when grouping topics for that audience.

After reading your post, that sounds right to me.

> In the past has documentation gotten bogged down in the details such as these? Would it be better to forge ahead so there is some documentation out there instead of promising that it is coming soon?

I agree that we should definitely avoid making the perfect (which is subjective and therefore unachievable) the enemy of the good--we can always edit and reorganize existing documentation later!

**Comment by tomgrandy** [comment]
@jackaponte - You are welcome and thanks for the feedback!

**Comment by Graham-72** [comment]
I had a look the other day at the section called 'Quick Start Guide' https://backdropcms.org/user-guide/quick-start and added some screenshots.

I think this section is just about what is needed to get new site-editors started.

Almost certainly it would need to be supplemented by some website-specific documentation, produced by the site designer/developer. For example, the site might use a theme that makes it look rather different.

**Comment by klonos** [comment]
> Luckily we can certainly anticipate and account for overlapping audiences in our approach here.

We can organize topics in an order that seems more reasonable as a workflow and have and "audience" taxonomy tag that is a multi-select checkbox. That way:

- docs can be marked for intended audience
- we can have a "browse site-building topics"/"browse content authoring topics" filter that "personalizes" that list for the current viewer.
- perhaps also include a difficulty rating in there somehow? Like allow people to vote by selecting the roles that describe them best and then a 5-star rating (with an optional feedback text field in order to see which parts of the documentation need to be improved).

**Comment by klonos** [comment]
> For example instead of:

> CONTENT EDITOR

> Logging In
> Creating Content
> Admin Bar

> The documentation would read:

> CREATING CONTENT

> Logging In
>Creating Content
> Admin Bar

I think that task-based documentation is better than audience-based too. The intended audience should be a documentation property (tag) - not defining the order/categorization of topics. It could be used for filtering topics. ...OR, we could offer two ways to browse documentation: one by tasks and one by audience.

**Comment by klonos** [comment]
> I agree that we should definitely avoid making the perfect (which is subjective and therefore unachievable) the enemy of the good--we can always edit and reorganize existing documentation later!

It is the enemy of *done* - not of good 😄

**Comment by jenlampton** [comment]
audience info should be hidden. People don't often self-identity, and if
they do they are likely to draw the lines in different places than we do
(for example, @docwilmot and @klonos still claim they are not coders).
Audience labels could end up causing more harm than good.

Audience should be important for us to think about while building, but
should not be important for those using the docs when we are done.

**Comment by docwilmot** [comment]
> Would it be better to forge ahead so there is some documentation out there instead of promising that it is coming soon?

Yes. Yes, indeed.
We already have several docs which are reasonably ready to publish IMHO. I would suggest next steps:
- @tomgrandy @jackaponte and anyone else go over the existing docs once more to ensure no glaring errors. Document (anywhere)
  - any errors (we can fix immediately)
  - any omissions (missing topics) (perhaps just add to the checklist in the first post)
  - any recommended improvements to the text
- @wesruv @klonos create screenshots (you've both volunteered before right?) as we go along. We dont need to wait for them.
- @tomgrandy propose a landing page TOC and any sublanding page in the style of the WP with annotations to show which docs your headings should link to. We'll argue about it briefly.
- Once we've argued enough (a day or two, no more), I'll paste that TOC on the current doc landing page and create the actual links
- We go live once @jenlampton clicks the button. I'd recommend we push live and fix as we go along.

**Comment by tomgrandy** [comment]
> Audience should be important for us to think about while building, but
should not be important for those using the docs when we are done.

@jenlampton - I agree that audience is important when approaching each topic area for it will guide the terminology used there to ensure that readability is recognized.  Since there is so often an overlap in what we do (some wearing all hats) in smaller shops, it would be difficult to separate the titles, agree.  However, a hierarchical order of difficulty can be adhered to.

**Comment by tomgrandy** [comment]
> @tomgrandy @jackaponte and anyone else go over the existing docs once more to ensure no glaring errors. Document (anywhere)
any errors (we can fix immediately)
any omissions (missing topics) (perhaps just add to the checklist in the first post)
any recommended improvements to the text

I'll go through the documentation and install Backdrop following the step-by-step and look for anything that is missing.  I have installed it, but not with documentation, so will do so from scratch with my newb hat on.

> @tomgrandy propose a landing page TOC and any sublanding page in the style of the WP with annotations to show which docs your headings should link to. We'll argue about it briefly.

Will put this together for discussion purposes and, hopefully, publication.

Let's do this!

**Comment by wesruv** [comment]
@docwilmot @klonos re:graphics

I specifically mean graphics that are used to describe abstract concepts, e.g. if we wanted a diagram that explained the flow of content from how it's stored in a node to view mode display settings to template... or something 😛 

Like these things I just made for a Lullabot article:
![vh](https://cloud.githubusercontent.com/assets/5607236/18966432/295313c6-864f-11e6-8826-54cf0890e269.png)

![viewport-height](https://cloud.githubusercontent.com/assets/5607236/18966440/2e174e68-864f-11e6-9eb4-ca47c04dfcec.png)

![vmin](https://cloud.githubusercontent.com/assets/5607236/18966445/312dec6a-864f-11e6-998f-3bbb3787f89c.png)

![vmax](https://cloud.githubusercontent.com/assets/5607236/18966450/37605316-864f-11e6-9537-6342ad7603fa.png)

**Comment by tomgrandy** [comment]
Nice work!  Screenshots are excellent for what already exists (especially as animated gifs to show quickly where something is located or how to do a simple task.  However, abstract concepts would require graphics and by what you have shown above, you have the mad skills!

**Comment by tomgrandy** [comment]
Do we follow a particular style guide with regard to text used on the website?

AP, APA, Chicago, Gregg ??

Also, have you adopted styles for use of capitalization in Titles and Callouts?

Lastly, are we using American English, British English, or International English?

My Virgo attributes are showing, I know, but want to make sure I don't have to backtrack here.

**Comment by docwilmot** [comment]
I vote that we allow @tomgrandy to make this decision on our behalf. Especially since I have no idea. 😄

**Comment by tomgrandy** [comment]
@docwilmot - That is mighty kind of you!  I think I'll refer to the AP Style Guide if I have a question regarding use since it is the standard for journalists worldwide.

Also, I'll create what is called a House Style Guide as I go.  An House Style Guide documents the styles adopted for a particular website so it maintains consistency across the board.  

Also, let's defer to American or International English since those seem to be adopted by most international corporations with regard to their web content.

I'll post a link to the House Style Guide as soon as I have enough listed.

Two beginning examples, based on reviewing how other sections were created, are:

**Titles**

1. Capitalize nouns, pronouns, adjectives, verbs, adverbs, and subordinate conjunctions.
2. Lowercase articles (a, an, the), coordinating conjunctions, and prepositions.

**_Sample: “Add-ons: Modules, Themes, and Layouts”_**

**Callouts**

1. Capitalize first letter and proper nouns.

**_Sample:  “Handbook for help with how to install Backdrop CMS”_**

**Comment by tomgrandy** [comment]
@docwilmot - Question regarding the Quick Start Guide followed by the User Guide.

There seems to be a great deal of repetition between what is presented in the Quick Start and then presented again in the User Guide sections.  It is confusing to read and then re-read the same later in the documentiation.  Would it be better for flow to keep the content listed once in the order presented?  

IHO, much of the Quick Start material should be presented in the section(s) it pertains to so not to cause confusion to the reader.

**Comment by docwilmot** [comment]
I feel the quick start is meant as a comprehensive but very low level overview. A pamphlet, rather than a booklet or a thesis. @Graham-72 particularly liked this.

**Comment by tomgrandy** [comment]
Got it!  Just wanted to ensure we weren't being overly repetitive.

**Comment by klonos** [comment]
> @docwilmot @klonos re:graphics ...

I'm all for graphics @wesruv 👍 

> I vote that we allow @tomgrandy to make this decision on our behalf. Especially since I have no idea. 😄 

I vote yes to that too. Especially since I don't have any idea + no time either 😄 

> **Titles**

> 1. Capitalize nouns, pronouns, adjectives, verbs, adverbs, and subordinate conjunctions.
> 2. Lowercase articles (a, an, the), coordinating conjunctions, and prepositions.

> **_Sample: “Add-ons: Modules, Themes, and Layouts”_**

> **Callouts**

> 1. Capitalize first letter and proper nouns.

> **_Sample:  “Handbook for help with how to install Backdrop CMS”_**

👍 

I would also say use italics for words withing sentences that are specific **named** parts of the UI. For example:

> Go to the *Manage blocks* page and click the *Add block* button on the *Header* region ... click the *Save configuration* button.

...but:

>To add menu items to the primary navigation menu of your website, use the Admin menu at the top of the page and navigate to *Structure* > *Menus* > *Primary navigation* > *Add link*

...or:

> To edit a page or post, click the *Edit* option available in the primary tabs section below the page title.

Notice how the "primary navigation menu", "Admin menu", "primary tabs" and "page title" are not in italics. These are actual things (parts of the UI) that we refer to, but there is no such text (as a label for example) in the page that the user can see. For example, if the user uses their browser search function (Ctrl+F, F3 or Command+F) to search for "page title", there will be no result returned. On the other hand:

> To customize the page title of your website, go to *Structure* > *Layouts* > *Default layout* > *Manage blocks*, then click the *Configure* button available in the *Page title* region.

...this time, the "Page title" part is in italics because it is an actual *labeled* thing on the page.

**Comment by klonos** [comment]
...I would also say use italics for words withing sentences that are specific **named** parts of the UI. For example:

> Go to the *Manage blocks* page and click the *Add block* button on the *Header* region ... click the *Save configuration* button.

...but:

>To add menu items to the primary navigation menu of your website, use the Admin menu at the top of the page and navigate to *Structure* > *Menus* > *Primary navigation* > *Add link*

...or:

> To edit a page or post, click the *Edit* option available in the primary tabs section below the page title.

Notice how the "primary navigation menu", "Admin menu", "primary tabs" and "page title" are not in italics. These are actual things (parts of the UI) that we refer to, but there is no such text (as a label for example) in the page that the user can see. For example, if the user uses their browser search function (Ctrl+F, F3 or Command+F) to search for "page title", there will be no result returned. On the other hand:

> To customize the page title of your website, go to *Structure* > *Layouts* > *Default layout* > *Manage blocks*, then click the *Configure* button available in the *Page title* region.

...this time, the "Page title" part is in italics because it is an actual **labeled** thing on the page.

**Comment by klonos** [comment]
...eventually, I would like us to switch those italics to have it so that the documentation text that is referring to UI elements like admin menu links, dropbuttons and form submit buttons is styled to resemble the style of those elements. Something like this:

![backdropcms org-awesome_docs-ui_elements_text_styled](https://cloud.githubusercontent.com/assets/2423362/19017932/7cfc39fa-889b-11e6-836b-e918dab948dd.png)

**Comment by klonos** [comment]
...some might find this too flamboyant, but it would solve the problem of the documentation gets too "wordy" (because it is meant for beginners). As users progress to more advanced level of knowledge, they can simply skim through the documentation and scan for the parts that stick out 😉

**Comment by jenlampton** [comment]
These styles should already be options in the rich-text editor. Feel free
to try them and see how it feels.

- -
Please excuse brevity and typos. Message typed with my thumbs from an Open
Source OS.

**Comment by klonos** [comment]
> These styles should already be options in the rich-text editor. Feel free to try them and see how it feels.

That is so cool! I didn't realize that has already been implemented. Thanx 👍

**Comment by tomgrandy** [comment]
Excellent editorial guides, @klonos !  Thanks @jenlampton for pointing out the styles are there.  Got sidetracked with a nice day for sailing Sunday but plan to get back to it this evening.

**Comment by olafgrabienski** [comment]
> I would like us to switch those italics to have it so that the documentation text that is referring to UI elements like admin menu links, dropbuttons and form submit buttons is styled to resemble the style of those elements.

Personally, I don't like the styles referring the UI elements:

- There are too many elements styled in a different way, which is not very readable. The text shouldn't be so 'busy', in my opinion.
- Many elements in the UI are uppercase. That's (more or less) fine in the UI but not in a running text.

I'd suggest to keep italics (most elegant), or to use bold (more clear imo). Following a short comparison:

- Go to the _Manage Blocks_ page of the Default layout and click the _Add Block_ button on the _Header_ region ...
- Go to the **Manage Blocks** page of the Default layout and click the **Add Block** button on the **Header** region ...

**Comment by jenlampton** [comment]
The styles are less important if we also have a screenshot. We just shouldn't require people to actually read the all the text. If we can tell the story with a screenshot instead of styled text, that's fine. If we don't have a screenshot, we just need to make sure people can still find the information they need by scanning.

**Comment by olafgrabienski** [comment]
I'd like to give feedback to the new structured documentation if needed, but I'm not sure about the current status of the content-related tasks. Any updates?

**Comment by serundeputy** [comment]
`508`

**Comment by tomgrandy** [comment]
I've been editing the documentation for spelling, grammar, punctuation, and usage errors.  My plan is to then go through step-by-step as a newb and see if the documentation makes logical sense in the order it is presented and take screenshots to be included as I move through it.  Not there yet, but do want to get it done as soon as possible.

**Comment by tomgrandy** [comment]
@serundeputy - Are you referring to 508 Compliance?  

_Section 508, an amendment to the United States Workforce Rehabilitation Act of 1973, is a federal law mandating that all electronic and information technology developed, procured, maintained, or used by the federal government be accessible to people with disabilities._

**Comment by serundeputy** [comment]
yes

**Comment by tomgrandy** [comment]
I see Colorbox is a module that is available for Backdrop.  Do you have it enabled on the Backdropcms.org site?  It would be ideal for the screenshots I put in the User Guide to be clickable to open a larger version since some of the text in a screenshot gets rather small.

Something akin to clicking on the image on this page:  http://www.alessioligabue.it/colorbox-image-load

**Comment by tomgrandy** [comment]
Also, are you documenting changes made to pages with notes in the Revision Information?  I went through and looked at some revised pages, but do not see notes as to what was changed.

**Comment by docwilmot** [comment]
@tomgrandy we dont have colorbox, but it would probably be a good idea. That would be @jenlampton though. 
And revisions weren't on (except maybe on random nodes) since this was just the initial build. Now that you're actually revising, perhaps we should.

**Comment by tomgrandy** [comment]
@docwilmot - Thanks!  Agree Colorbox would be nice to see screenshots better.  
@jenlampton - Please advise if Colorbox is possible to allow for users to click on images to enlarge.
@docwilmot - I have started to note what changes I've made to the User Guide in revision info.

**Comment by jenlampton** [comment]
> Please advise if Colorbox is possible to allow for users to click on images to enlarge.

Yes, this seems like a reasonable request. I'll look into it on Thursday :)

**Comment by klonos** [comment]
Related: https://github.com/backdrop/backdrop-issues/issues/1465

...for starters, we could port https://www.drupal.org/project/diff and install it in b.org, but ideally we should implement something like this in core:

![wp-browse_content_revisions](https://cloud.githubusercontent.com/assets/2423362/12071293/73c3fe52-b0f8-11e5-8c20-15398c6daf08.gif)

**Comment by tomgrandy** [comment]
It would be awesome to have the diff module installed so we can track revisions without manually typing every change into the Revision Information.  I've been putting general comments in and when I alter content, I paste that into the RI for reference.

**Comment by tomgrandy** [comment]
@klonos - What did you use to create the animated gif above?

**Comment by jenlampton** [comment]
@tomgrandy it doesn't look like we have a diff module for backdrop yet. Maybe create an issue over at https://github.com/backdrop-ops/contrib/issues and see if anyone is working on it yet (or wants to?).  We do have the diff engine (and colors!) in core already, so I think it shouldn't be too much trouble.

**Comment by jenlampton** [comment]
I'm not happy with the current colorbox module. Giving it a rewrite for backdropcms.org.

**Comment by jenlampton** [comment]
okay, fixed up colorbox (https://github.com/backdrop-contrib/colorbox/pull/10) and pushed it up. I'm not sure how you're going to add the necessary classes on the inline images, but let's see if we can't get it working.

**Comment by tomgrandy** [comment]
@jenlampton - Thank you for taking the time to tweak colorbox and push it up.  Will give it a shot today to see what I can do.  

Also, will post to contrib/issues to see if there is anyone working on a diff module.

**Comment by klonos** [comment]
@all sorry guys for not being very responsive lately. I cannot even begin to explain how busy I am these days.

> It would be awesome to have the diff module ported and installed so we can track revisions without manually typing every change into the Revision Information.

👍 1 million.

> @klonos - What did you use to create the animated gif above?

http://blog.bahraniapps.com/gifcam but there's also http://www.cockos.com/licecap that works on OSX

**Comment by tomgrandy** [comment]
@klonos - No worries.  Thanks for taking the time to respond.  I posted a request to see if anyone was working on, or thinking about working on, a diff module on github at: 

https://github.com/backdrop-ops/contrib/issues/182

Will see if that garners any responses.

Thanks for the gifcam link.  Will give it a shot 👍

**Comment by tomgrandy** [comment]
@wesruv - Can you take a look a the source for the colorbox-inline link that open in a colorbox, but does not display.  Not the easiest way to do this, but if i can get an href that works, I can swap filenames in for screenshots to be enlarged with colorbox.

<a class="colorbox-inline" href="https://backdropcms.org/files/inline-images/login.png"><img alt="Login" src="https://backdropcms.org/files/inline-images/login.png" title="Login" width="500" /></a>

Appears as a test here:  https://backdropcms.org/node/468/

**Comment by jenlampton** [comment]
Could we give you a style, like we have for the navigation and submit buttons, maybe?

**Comment by tomgrandy** [comment]
Style would be fantastic!  Just so I have a way to identify those images that should utilize colorbox.

Thanks

**Comment by tomgrandy** [comment]
Rest assured that I **_am_** that user who is using your site for the first time to understand how to install and use Backdrop because that is exactly what I am doing.  And the order of presentation is very confusing.

The content is solid and most everything I need to know is already there, it is just the order makes me have to hunt for what I need and that is not good.

Therefore, I propose the following that is not intended to start any heated discussion and feel free to tell me know, but if you wanted a user group to test on, that is me.

So for the good of the cause and the new user, I propose I do the following:

1) The **About Backdrop** section does not belong in the **Handbook** but should stand on its own accord as a link above the **Handbook** link.  It is important background information, vitally important to someone looking to adopt, but it is not on _how_ to use Backdrop as one would expect in a Handbook.

2) You have a User Guide and a Developer Guide in two different places and some of my needs overlap both for basic knowledge like installing backdrop.  We can't assume some IT department is going to do it for me, especially if I have my own shop.  The two should merge with a hierarchy of order from the simple to the complex such as:

**Installing Backdrop**

- System Requirements
- Installation Instructions

followed by 

**The Basics:**

- How to Log In
- Admin Bar
- Content
- Comments
- Contact Form
- Search

**Visual Aspects of The Site:**

- Themes
- Layouts
- Blocks
- Menus

**Customizing Content:**

- Content Types
- Fields
- Taxonomy
- Views

**Extending Functionality**

- Modules
- URL Aliases
- URL Redirects

**User Management**

- Creating Users
- Identifying Roles
- Customizing Permissions

**Developer Tools**

- Upgrading From Drupal 7
- Backdrop Upgrades
- Submitting Pull Requests
- Developer Documentation
- Backdrop API

^^ Everything in the Developer Tools section is something I would probably pull in another person to work on,  Everything above that is what I need to know to build and manage a site on my own.

Be kind.  I'm trying to make this work and, for the most part, all the pieces are in place.

**Comment by tomgrandy** [comment]
Also, check the site out on mobile with regard to where the menus end up (bottom).

Granted, we are probably all looking at the site on dual monitors, but mobile rules and having the menu at the bottom makes it hard to navigate.

Should choose either the l-drawer off the Handbook link or the l-sidebar inner, but not both for mobile navigability.

Wordpress Codex is not acceptable on mobile  https://codex.wordpress.org/

However, https://learn.wordpress.org is mobile friendly with a menu that is operable.

FWIW

**Comment by jenlampton** [comment]
One correction to your thinking: the handbook and the user guide are
separate concepts with no relation to one another.

The handbook is a document someone new to backdrop can read (every page)
front-to-back to understand the what/why of backdrop. *Not* the how.

The how of Backdrop is in the user guide. The user guide will have its own
place in the menu (under support) separate from the handbook. The
documentation will be too extensive and feature-rich to be considered a
handbook. It's a reference guide.

Ignore the about section and everything in the handbook while we are
working on the user guide, and we can revisit the handbook next, if you are
up for it :) (no pressure!) Some links may be duplicates only because we
had no user guide before now. Those could be deleted / redirected as
necessary, but don't let them confuse the point of the handbook.

I like your suggestions about reworking the order of the user docs. It
Looks great to me!

The mobile navigation is a bit of a hack, and could use reworking in
general. We have some work to do :/

**Comment by tomgrandy** [comment]
@jenlampton - Thanks for the clarification.  I was confused by the overlap of "how" content that appears in the Handbook pop-out menu (ie - User Guide, Developer Guide). 

If you are okay with me repurposing parts what has been written in the categorized order I presented, I'll copy them into the hierarchical flow from simple to difficult in the User Guide.

I'm always up for a challenge and if you all like what I do with the User Guide, then we can take a look at the Handbook which should, as you pointed out, give the new user a full background of the what/why and leave the how out.

Didn't want to step on any toes.  Figure it is better to ask ahead of time than to ask for forgiveness after the fact.

**Comment by klonos** [comment]
> Also, check the site out the BackdropCMS.org site on mobile with regard to where the menus end up (bottom).

Yep, I've filed https://github.com/backdrop-ops/backdropcms.org/issues/292 some time ago about that.

> (Is this where I am invited to no longer work on documentation?)

I doubt that this will ever happen for either you or anybody else. At least not for making suggestions and having a constructive conversation about improving the website and documentation.

**Comment by klonos** [comment]
> I like your suggestions about reworking the order of the user docs. It Looks great to me!

Same here. I might have a few suggestions/opinions of my own, but no time to elaborate on that right now. Other than the Handbook vs User Guide (on which matter I'm with @jenlampton by the way), everything else seems like a very reasonable improvement to me. Go ahead!

**Comment by tomgrandy** [comment]
Thanks @klonos for your feedback and encouragement.

I look forward to hearing your suggestions/opinions whenever you have time.

**Comment by klonos** [comment]
> Didn't want to step on any toes.

You are not stepping on any toes - you are helping a big deal.

> Figure it is better to ask ahead of time than to ask for forgiveness after the fact.

We do both in Backdrop-land. If something is well-intended, it simply does not matter.

**Comment by tomgrandy** [comment]
@quicksketch - I'm assuming you are running a caching server.  What is the refresh on that?

**Comment by olafgrabienski** [comment]
@tomgrandy - Do you intend to write a section about using Backdrop in other languages (localization)? Didn't saw it in https://github.com/backdrop-ops/backdropcms.org/issues/263#issuecomment-253951250. In my opinion we should add such a section, and I'm interested to help.

And regarding your outline proposal: Would such a section fit under "Basics" or better under "Extending functionality"?

**Comment by quicksketch** [comment]
> @quicksketch - I'm assuming you are running a caching server. What is the refresh on that?

It should just be following the Backdrop cache settings, which I believe is a 5 minute cache.

**Comment by tomgrandy** [comment]
My apologies for being AWOL lately.  Have been extremely busy at work trying to get a couple sites launched and wanted to get them finished so I could come to BadCamp without workplace baggage following me.

So, it doesn't look like I made my "by BadCamp" self-imposed deadline of getting the User Guide completed.  Will work on it tonight and on my way to BadCamp if Frontier's onboard WIFI is decent.

Hope the training went well today.

**Comment by tomgrandy** [comment]
@olafgrabienski - I know next to nothing about Backdrop in other languages and welcome the help.  Would extend the audience of those who don't create sites in English only.

**Comment by tomgrandy** [comment]
@docwilmot @Graham-72 @jenlampton @klonos 

Sitting in an airport waiting for my flight to San Francisco so have been working on the site.

Would like to use a standardized Style Guide.  Please look over the this page and see if it fits what you have in mind for keeping the documentation similar:

https://backdropcms.org/user-guide/website-style-guide

Thanks!

**Comment by docwilmot** [comment]
Looks good. Re callouts, we could use different classes of callouts, like "pro tips" vs "important!" etc. These would need to be styled differently. Should this style guide reflect that? Hope you understand.

**Comment by tomgrandy** [comment]
@docwilmot - I was referring to just the text field "Callout" on the documentation page which doesn't look like I can add classes to the field.  Are you referring to using Callouts within the body itself to draw attention to differentiated levels of difficulty?

**Comment by docwilmot** [comment]
Ok didn't realise. I'm sitting in a healthcare conference on my phone myself, so didn't check. 
But yes, that's what I meant. I think there were styles in ckeditor for callouts off that sort already, I think. 
What's your opinion on those?

**Comment by tomgrandy** [comment]
@docwilmot - There is an item in the Styles menu named "Callout" but it doesn't do anything that I can tell.  Perhaps @wesruv can shed some light on that one.

**Comment by olafgrabienski** [comment]
@tomgrandy Alright, I'll try to write about using Backdrop in other languages. (Due to my modest English language skills it'll need some proofreading.) Currently I'm pressed for time, can't start immediatly but will come back to it soon.

And how would the collaboration work technically? Do I need a particular role on b.org to edit the User guide?

**Comment by tomgrandy** [comment]
@olafgrabienski - I would be happy to copyedit anything you put together.  As for what role you need, do you currently have access to edit the backdropcms.org site?

**Comment by olafgrabienski** [comment]
@tomgrandy Re edit access: No, I don't think so.

@jenlampton May I get a role on backdropcms.org that allows me to edit the User guide?

**Comment by tomgrandy** [comment]
@olafgrabienski - My mistake.  You mean localization of Backdrop making the CMS administration in other languages, correct?

**Comment by tomgrandy** [comment]
https://backdropcms.org/user-guide/backdrop-other-lanuages

**Comment by olafgrabienski** [comment]
@tomgrandy Yes, localization. Btw, typo in link: "lanuages"

**Comment by olafgrabienski** [comment]
@tomgrandy Had a quick look at the Website Style Guide. Sorry to say, but I'm still not convinced of the styles which mimic the UI. As I said in https://github.com/backdrop-ops/backdropcms.org/issues/263#issuecomment-251807415, these styles aren't very readable in a text. What do you think about it?

**Comment by tomgrandy** [comment]
@olafgrabienski I do like the italics best.  With screenshots, I can ensure that most of what people are reading is better understood by combining text explanation with images.  Currently, the User Guide uses many styles and would like to agree that we use either use the UI-type styles or we don't.  Whatever needs to be consistent across the board.  If we can get some agreement that would be ideal so I can correct it as I go through putting screenshots in place.

**Comment by tomgrandy** [comment]
@olafgrabienski @jenlampton 
Here is a page that uses only text, italics, bold, links, and screenshots to explain what options are available on a Content Form.  Does this look less busy?

**Comment by docwilmot** [comment]
You forgot to post the page?

**Comment by klonos** [comment]
@olafgrabienski with regards to being able to edit documentation, I just granted you the "editor" role which should give you permissions to do that. Let me know if that works for you.

With regards to proofreading, I will do my bit as time permits and with regards to the multi-lingual topics of the User Guide, I can also help.

Now, as for the styles, sorry but I am with @tomgrandy and the others on that one. I think that having the text that refers to buttons and navigation for example resemble the respective default admin UI elements helps a great deal. Especially for people that do not need to read the fine lines and are simply looking for something to quickly go through. This way, the entire, detailed literature is there for beginners, but the more advanced users that are looking for quick tips can get the info they need in a fast read through.

**Comment by klonos** [comment]
@jenlampton, now that we have a small number of people editing the documentation nodes, should we prioritize https://github.com/backdrop-ops/backdropcms.org/issues/240 ??

**Comment by klonos** [comment]
...also, should we add some more options in the "Status" field of documentation nodes? I was thinking a "Needs proofreading" and "Needs screenshots" for starters. Also, perhaps convert it into a checkbox list instead?

**Comment by tomgrandy** [comment]
@docwilmot - Sample page = https://backdropcms.org/user-guide/content-form

@klonos - Great idea for Status field and thanks for setting up @olafgrabienski !

@olafgrabienski - Languages, geez.  Looks like you or someone corrected it for me.

Just got into my house for Bad Camp tomorrow.  Long day.  Need sleep.  Will write more tomorrow.

**Comment by olafgrabienski** [comment]
@klonos @tomgrandy Thanks for your replies and setting up the role on b.org. I'm more or less offline this weekend, will try and answer monday or tuesday.

**Comment by tomgrandy** [comment]
Whoa!  Diff Module is now available in Backdrop as of Oct 22? 

@Graham-72 / Thanks!  Will put it on my sandbox and see if we can use it on B.org for tracking changes to documentation.  Would make me feel a lot better about having a history of what was changed in case someone wants it returned to a previous version.

**Comment by Graham-72** [comment]
@tomgrandy - sorry, a bit premature, there is a problem with my porting, and in trying to solve one thing I have introduced another problem. Rather embarassing and a bit frustrating after seeing it start to work!

**Comment by tomgrandy** [comment]
@Graham-72 - No worries!  Got my attention there.  Already installed and scratched my head a bit.  Will uninstall and wait for your new and improved version.  Still, thanks!

**Comment by klonos** [comment]
@Graham-72 👍

**Comment by tomgrandy** [comment]
@jenlampton As I put the screenshots into the User Guide, can I have access to enable an image to use Colorbox as a style?  Can use it on my sandbox, but not enough permission to customize it on BackdropCMS.org.

**Comment by jenlampton** [comment]
sorry @tomgrandy I badcamp got me again. if colorbox is working in your sandbox we can enable on live too.

**Comment by wesruv** [comment]
> There is an item in the Styles menu named "Callout" but it doesn't do anything that I can tell. Perhaps @wesruv can shed some light on that one.

@tomgrandy 
It sounds like we want to style callouts so they have a little more visual dominance than the rest of the copy? If there's a link with an example I can whip something up! Or let me know if I'm not understanding the question?

**Comment by tomgrandy** [comment]
@jenlampton & @wesruv - Just got home from a delayed red-eye and went straight to work from the airport.  Need to get caught up with all the tickets that piled up while I was gone then will go in and see how Colorbox can best work with my sandbox site and then let you know what is needed to get it to affect the images on Backdrop's flagship site.  Thanks for getting back with me.

**Comment by docwilmot** [comment]
@tomgrandy now that we're at 102 comments on this issue, maybe time to take stock? What are we doing right now? Whats left to plan and to do?

One of the main issues I see is that we have 24 really big topics which will each need screenshots; hows that going to work out? I would suggest we tackle a few topics at a time, and you let us know which screenshots you need and we divvy them up; otherwise this will be forever.
> re Does this look less busy? https://github.com/backdrop-ops/backdropcms.org/issues/263#issuecomment-255491455 

Yes it does look better; I think I would side with @olafgrabienski and say the current formatting is too distracting (the black background on the breadcrumb "Navigation style" especially). Your style guide looks great without that.

I think we should next agree quickly on the Style Guide (I vote to nix the navigation background though). 

I also would suggest that we make side issues for things like adding Diff and Colorbox and Languages and so on and get these done in parallel or ideally later, as we seem to be bogging down a bit. Lets stick to this issue which is just creating great docs which have been proofread by somebody and ready to post. We can decorate and embellish later IMHO.

In summary, could we limit this issue to: 
- do we have all the articles we need
- are they in the proper categories/hierarchy
- do we have proper landing pages for users to browse
- are they proofread and simplified and acceptable to @tomgrandy to mark complete

I think once those are satisfied, we can publish so our users can at last get *some* docs.
[Perfect is the enemy of good](https://github.com/backdrop-ops/backdropcms.org/issues/263#issuecomment-250219865)

**Comment by tomgrandy** [comment]
@docwilmot - Didn't mean to muddy the waters.  But at the same time, didn't want to have to go back in and change all the screenshots if we were going to use a method that would allow people to see the details of images (duplication of effort).

I will change the Style Guide to keep the text simple.

I will mark complete those pages I have proofread.

I will change the hierarchical order of the pages to match the outline 

Feel free to start posting screenshots on pages.  Try to keep them at a consistent 600px wide.

As for making side issue, all for it, but don't know this well enough to start them in the right place.  If you could make one for diff, I'll make one for colorbox and languages based on the diff that you start.

Other than that, my plan was to continue moving through the context and uploading screenshots as I do what the user guide explains to ensure it is correct with the latest version.

Hope that sounds like a plan.

**Comment by docwilmot** [comment]
@tomgrandy we all muddied the waters, not you. The summary of my rant above is essentially: lets get the *text* ready first. 

The other stuff doesn't have to be parked, but let's leave them as secondary since we could have a perfectly good user guide without images, colorbox, diff and else. 

For example @klonos could create the screenshots you need at full size and store them labelled in GDocs in folders corresponding to the articles, until Colorbox is ready, then himself or  you and/or someone(s) else can insert them then. Same with Diff: @jenlampton can just make sure that revisions are enabled, until Diff is ready, then she can install and plug it in. 

But for now we just fix the text.

Hope you understand what I mean?

**Comment by olafgrabienski** [comment]
@klonos Thanks for the editor role. Just tested it quickly, works fine so far!

**Comment by tomgrandy** [comment]
@jenlampton - Having no success getting inline images to open in Colorbox.

Works great with attached images Manage Display / Image / Format = Colorbox

Not working on the inline images.  Since most of what I will be doing is multiple inline images in a page of instructions, am scrapping the idea of utilizing Colorbox at this point.

Seemed like a good idea.

**Comment by jenlampton** [comment]
There may need to be some colorbox/ckeditor integration we need to add.

- - -
Jennifer Lea Lampton | 510-747-8230
Jeneration Web Development
Backdrop CMS & Drupal
www.jeneration.com

**Comment by tomgrandy** [comment]
@jenlampton - As mentioned in today's meeting, please change the "previous" and "next" navigation link at the bottom of each page astay at the top level and not go into the secondary-level Deep-Dive sections.

**Comment by tomgrandy** [comment]
@jenlampton - Also, you currently have Status tags of Incomplete and No Known Problems.  Do we want to have tags like "Beginner" and "Advanced" as Gregory pointed out in today's meeting?

**Comment by jenlampton** [comment]
@tomgrandy Level taxonomy added, see: https://backdropcms.org/user-guide/creating-new-content-type  Working on the next/previous links now.

**Comment by jenlampton** [comment]
@tomgrandy next/previous links modified to ignore child pages: https://backdropcms.org/user-guide/content-types

**Comment by tomgrandy** [comment]
@jenlampton - Adding "Beginner" or "Advanced" tags to all pages and the Next / Previous links are so much more logical for moving through the basic documentation.  Big thanks!

**Comment by tomgrandy** [comment]
Only two more sections to go and the User Guide is complete.  Would someone be willing to review and put any necessary screenshots (500px wide) on this page:

https://backdropcms.org/user-guide/project-installer-and-updates

This is one that doesn't seem to follow the current version or is missing in the version I have installed.

Thanks!

**Comment by tomgrandy** [comment]
Filesystem is writable and I can install modules easily (love it).

I'll go through the project-installler-and-updates again, but something doesn't match.

Oh, and @jenlampton , you might want to remove your phone number from the comment above.

Unless you want it there.

**Comment by tomgrandy** [comment]
I see where the page falls through the cracks for me.

It is an excellent description of "how" the project installer works at the beginning, in great detail, that a person who needs to know "how-to" would not be looking for.

Would it be okay for me to rewrite the page with the "how-to" install modules, themes and templates without explaining the magic behind scenes:  

"Installer is closely coupled with the several core and remote components and requires online access to work. The main remote service is BackdropCMS’ Project Browser Server which communicates with Installer and serves lists of available projects for download. The following diagram illustrates the connections between the components required for a working Installer."

The above is confusing if this is aimed at someone who is new to CMS and Backdrop in particular.

It would make a great deep-dive, but not a how-to.

Make sense?  Here I was looking for a Project Installer.  That term does not appear anywhere in the admin bar, hence the confusion that I didn't have it.

**Comment by jenlampton** [comment]
Yes, please go ahead :)

- -
Please excuse brevity and typos. Message typed with my thumbs from an Open
Source OS.

**Comment by olafgrabienski** [comment]
@tomgrandy Good to hear of the progress! At the moment, it makes no sense to update my page on Using Backdrop in different languages than English, because it's already difficult to install Backdrop in other languages, see https://github.com/backdrop/backdrop-issues/issues/2329.

I however want to come back as soon as possible to the topic but was wondering if the relevant page of the user guide has been deleted. Can't find it anymore.

**Comment by docwilmot** [comment]
@tomgrandy had a look at the progress so far, great work, hats off to you sir.
Do we need to consider section anchor links for the long pages? In case someone needs to link to the `Contexts` section within `Deep Dive: Advanced Layout Options` for example. We have some really long pages.
Is there anything we/I need to do to help finish?

**Comment by tomgrandy** [comment]
@jenlampton - I'll remake the page and put the background information on a deep-dive page if someone wants to peek behind the curtain.

@olafgrabienski - Sorry to hear the Backdrop in different languages isn't quite ready for prime time.  I did delete the page out of documentation so an empty page wasn't appearing in the menu now that the User Guide is about to go live..  Don't have the option to put something in an unpublished state.

@docwilmot - Thanks!  I'll take a look at the length of pages and see if I can break them up a bit.  If not, anchor tags would be a good option.  The Deep Dive pages are probably more apt to need something like this.  I'm hoping to finish the guide this weekend so it can be released to the wild.  Once I have it in what I think is a finished state if you wouldn't mind giving it a review for anything I may have missed and if you are willing to put the anchors on pages you feel are too long that would be awesome.

**Comment by klonos** [comment]
>Oh, and @jenlampton , you might want to remove your phone number from the comment above.

>Unless you want it there.

Yeah, don't think she noticed your comment @tomgrandy and I don't think she want's there because IIRC it was @quicksketch that was faking to add that to a node or something during the BADCamp training and from her reaction, I gather she was not into it 😄 ...removed that for you @jenlampton (but feel free to add it back in case you had it there intentionally)

**Comment by klonos** [comment]
...not sure I can make time for proofreading before you guys get to publish the docs, but I'll do that anyways at some point. Will also try to make time for screenshots, but cannot promise.

> @tomgrandy had a look at the progress so far, great work, hats off to you sir.

...second hat off here 👍 

I like the status indicator for incomplete documentation, but I think that it would be great to have a way for people (the regular, unregistered site visitors that read through it) to provide feedback in the form of "requires update" or "inaccurate" or even "typo" along with a short text like "I see no xyz button like the screenshot in the documentation". If that came out as an email to say @docwilmot, @tomgrandy and myself (the "documentation squad" + the "screenshot guy") so that we can take action, then that would be great ...I mean if the rest of the gang does not object to receiving such notifications and provided we get some decent anti-spam measures in place.

**Comment by tomgrandy** [comment]
Documentation is now live on BackdropCMS!  Very cool.

I'm still not 100% sure that I like the images maintaining a width of 600px because they are hard to read.  Does anyone else think that is the case?  If so, I can try bumping them up to 700-750px but then the question becomes how much real estate are we willing to dedicate to representing the interface or is it good enough as is?  

It would be easy for me to blow through documentation and resize to a consistent width that is easier to read since we cannot click to view the image.

If anyone else doesn't have additions / corrections, should we close this issue?

**Comment by docwilmot** [comment]
Congratulations Backdrop!
Well done @tomgrandy 
Have completely abandoned colorbox?

**Comment by jenlampton** [comment]
Colorbox is still on the table. We should figure out how to set that up for contrib anyway.

**Comment by alexfinnarn** [comment]
I'm not sure where to put this, but in #55, PHPStorm was mentioned. PHPStorm is a HUGE part of my DX and helps me catch a lot of bugs before I even run the code. 

In https://github.com/backdrop/backdrop/pull/2244#issuecomment-406795214, the multiline comment vs. docblock is easily noticable and the IDE helps to generate the docblock if one doesn't exist.

There is also a place to define code syntax standards. I often times select all and then automagically reformat things to Drupal's standards. 

https://www.jetbrains.com/buy/opensource/ Apache Foundation licenses.

**Comment by jenlampton** [comment]
@alexfinnarn we have a few PHPstorm licenses available for contributors to Backdrop for free. Anyone who is interested should email info@backdropcms.org for a license key!

**Comment by oadaeh** [comment]
Also, as an FYI, NetBeans can do the same things Alex said PHPStorm does. I suspect many IDEs can, if properly configured.
